core/src/test/java/org/mule/transport/polling/WatermarkTestCase.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.polling;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+import org.mule.transport.polling.watermark.selector.MaxValueWatermarkSelector;
+import org.mule.transport.polling.watermark.selector.MinValueWatermarkSelector;
+
+public class WatermarkTestCase {
+	
+	@Test
+	public void testMaxValueSelectedInWatermark() {
+		MaxValueWatermarkSelector selector = new MaxValueWatermarkSelector();
```

Comment:
None of these tests reproduce the condition you want to test, because you're using the Integer class which compareTo() method returns -1; 0; or 1